baseline/client/js/jspsych-spatial-orientation.js

```diff
@@ -0,0 +1,142 @@
+                ctx.lineTo(
+                    centerX + options.radius * Math.cos(pointerAngle + Math.PI/2),
+                    centerY - options.radius * Math.sin(pointerAngle + Math.PI/2)
+                );
+                ctx.stroke();
+                ctx.fillText(
+                    options.pointerText,
+                    centerX + (options.radius+40) * Math.cos(pointerAngle + Math.PI/2),
+                    centerY - (options.radius+40) * Math.sin(pointerAngle + Math.PI/2)
+                );
+            });
+        };
+        let running = true;
+        canvas.addEventListener("mousemove", e => {
+            if (running) {
+                draw(pointerAngleFromMouseEvent(e));
+            }
+        });
+        canvas.addEventListener("click", e => {
+            if (running) {
+                running = false;
+                // build data
+                const responseAngle = pointerAngleFromMouseEvent(e);
+                const clickData = {
+                    responseAngle: responseAngle,
+                    targetAngle: options.targetAngle,
+                };
+                // draw target pointer
+                ctx.setLineDash([]);
+                ctx.strokeStyle = "rgba(255, 0, 0, 1)";
+                ctx.beginPath();
+                ctx.moveTo(centerX, centerY);
+                ctx.lineTo(
+                    centerX + options.radius * Math.cos(options.targetAngle + Math.PI/2),
+                    centerY - options.radius * Math.sin(options.targetAngle + Math.PI/2)
+                );
+                ctx.stroke();
+                // call onClick
+                options.onClick(clickData);
+            }
+        });
+        draw(0);
+    };
+
+    plugin.trial = (displayElement, trial) => {
+        // build and show display HTML
+        {
+            let html = "";
+            html += `<div id="jspsych-spatial-orientation-wrapper">`;
+            html +=     `<div id="jspsych-spatial-orientation-scene">${trial.scene}</canvas>`;
+            html +=     `<canvas id="jspsych-spatial-orientation-icirc" width="500" height="500"></canvas>`;
+            html += `</div>`;
+            displayElement.innerHTML = html;
+        }
+        const icirc = document.getElementById("jspsych-spatial-orientation-icirc");
+        // check canvas support
+        if (!icirc.getContext) { return; }
+        const start = performance.now();
+        plugin.buildIcirc(icirc, {
+            radius: 150,
+            centerText: trial.centerText,
+            topText: trial.topText,
+            pointerText: trial.pointerText,
+            targetAngle: trial.targetAngle,
+            onClick: clickData => {
+                // build data
+                const rt = performance.now() - start;
+                const data = {
+                    ...clickData,
+                    rt: rt,
+                };
+                // finish trial
+                setTimeout(() => { jsPsych.finishTrial(data); }, 1000);
```

Comment:
I didn't see anything in the specs about time limits on individual trials (just on the experiment overall). Did I just miss it, or is this some temporary testing feature? It looks like trials are supposed to end when the user presses the enter key, which I'm assuming you haven't gotten to yet.

Reply:
Now that I've played with it I see that you're finishing the trial on click, not on enter. I like that better, so as long as it's ok with Hyun Joo that's great. Is there any reason to wait a second after the click to finish it, though?

Reply:
The delay is there to show the `targetAngle` red line!